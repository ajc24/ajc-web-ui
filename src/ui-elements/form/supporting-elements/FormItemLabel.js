import React from 'react';
import PropTypes from 'prop-types';
import '../../../css/common.css';
import '../../../css/form-item.css';

/**
 * The FormItemLabel component acts as the label for all form items.
 * This label supports both standalone form items and those rendered inside a dialog.
 */
class FormItemLabel extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    let labelStyling = 'ajc-row ajc-flex-justify-left ajc-spacing-bottom-small ajc-spacing-bottom-reduced';
    if (this.props.isDialogFormItem === true) {
      labelStyling += ' ajc-label-dialog';
    } else {
      labelStyling += ' ajc-label-standalone';
    }
    if (this.props.additionalUpperSpacing === true) {
      labelStyling += ' ajc-spacing-top-large';
    }
    return (
      <React.Fragment>
        <label htmlFor={this.props.id} className={labelStyling}>{this.props.labelText}</label>
      </React.Fragment>
    );
  }
}
FormItemLabel.propTypes = {
  /** The unique identifier for the form item that this label is linked to. */
  id: PropTypes.string.isRequired,
  /** The label text content for the text input component. */
  labelText: PropTypes.string.isRequired,
  /** Switch for adding more spacing above the label. Use this to separate the label from other items above it. */
  additionalUpperSpacing: PropTypes.bool,
  /** Switch for stating whether this form item appears in a dialog or not. By default this is set to false. */
  isDialogFormItem: PropTypes.bool,
};
FormItemLabel.defaultProps = {
  additionalUpperSpacing: false,
  isDialogFormItem: false,
};
export default FormItemLabel;
