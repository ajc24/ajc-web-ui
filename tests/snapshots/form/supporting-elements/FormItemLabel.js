import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import FormItemLabel from '../../../../src/ui-elements/form/supporting-elements/FormItemLabel';

describe('FormItemLabel', () => {
  describe('Default props and rendering - Standalone component, no additional spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FormItemLabel id="form-item-label-id" labelText="Test Form Item Label Text:" />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Dialog component with additional spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FormItemLabel id="form-item-label-id" labelText="Test Form Item Label Text:" additionalUpperSpacing={true} isDialogFormItem={true} />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
