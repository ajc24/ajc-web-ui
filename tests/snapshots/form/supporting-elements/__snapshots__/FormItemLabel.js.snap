// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItemLabel Default props and rendering - Standalone component, no additional spacing verifies the snapshot for the component 1`] = `
<label
  className="ajc-row ajc-flex-justify-left ajc-spacing-bottom-small ajc-spacing-bottom-reduced ajc-label-standalone"
  htmlFor="form-item-label-id"
>
  Test Form Item Label Text:
</label>
`;

exports[`FormItemLabel Transferred props and rendering - Dialog component with additional spacing verifies the snapshot for the component 1`] = `
<label
  className="ajc-row ajc-flex-justify-left ajc-spacing-bottom-small ajc-spacing-bottom-reduced ajc-label-dialog ajc-spacing-top-large"
  htmlFor="form-item-label-id"
>
  Test Form Item Label Text:
</label>
`;
